contracts/VeJoeStaking.sol

```diff
@@ -178,19 +286,33 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
 
         UserInfo memory user = userInfos[_user];
 
-        // Calculate amount of pending veJOE
+        // Calculate amount of pending base veJOE
         uint256 _accVeJoePerShare = accVeJoePerShare;
         uint256 secondsElapsed = block.timestamp.sub(lastRewardTimestamp);
         if (secondsElapsed > 0) {
             _accVeJoePerShare = _accVeJoePerShare.add(
-                secondsElapsed.mul(veJoePerSec).mul(ACC_VEJOE_PER_SHARE_PRECISION).div(VEJOE_PER_SEC_PRECISION)
+                secondsElapsed.mul(veJoePerSharePerSec).mul(ACC_VEJOE_PER_SHARE_PRECISION).div(
+                    VEJOE_PER_SHARE_PER_SEC_PRECISION
+                )
             );
         }
-
-        uint256 pendingVeJoe = _accVeJoePerShare.mul(user.balance).div(ACC_VEJOE_PER_SHARE_PRECISION).sub(
+        uint256 pendingBaseVeJoe = _accVeJoePerShare.mul(user.balance).div(ACC_VEJOE_PER_SHARE_PRECISION).sub(
             user.rewardDebt
         );
 
+        // Calculate amount of pending speed up veJOE
+        uint256 pendingSpeedUpVeJoe = 0;
+        if (user.speedUpEndTimestamp != 0) {
```

Comment:
There is an issue, if we change the `speedUpDuration`, current user that have a speedup won't have any benefit for it.
If we want to account for this, I suggest we add a `startSpeedUpTimestamp`, and before anything we test that `endSpeedUpTimestamp - startSpeedUpTimestamp == speedUpDuration`, if it's not the case, we change `endSpeedUpTimestamp` to account for the new changes.
Issue is that if we reduce the duration, then some users may have harvested before and thus would have had some `speedUp` rewards that they won't have if they'd have harvested only after the changes

Reply:
As per conversation in discord, aligned on making `speedUpDuration` constant